proto/registry.proto

```diff
@@ -17,3 +17,37 @@ message Manifest {
   // CAS anymore.
   repeated build.bazel.remote.execution.v2.Digest cas_dependencies = 4;
 }
+
+message Credentials {
+  string username = 1;
+  string password = 2;
+}
+
+message Platform {
+  string arch = 1;
+  string os = 2;
+  string variant = 3;
+}
+
+message GetOptimizedImageRequest {
+  // The target image, including repo (e.g. docker.io/busybox:latest)
+  string image = 1;
+
+  // If an image has multiple variants, this selector is used
+  // to select a single variant.
+  Platform platform = 2;
+
+  // Optional credentials for accessing the target image.
+  Credentials image_credentials = 3;
+}
+
+message GetOptimizedImageResponse {
+  string optimized_image = 1;
```

Comment:
same here -- it's nice to have examples of each field to help the reader understand what they're getting